src/select/SelectInput.ts

```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { BoxHTMLProps } from "reakit/ts/Box/Box";
+import { SelectStateReturn } from "./SelectState";
+import { createHook, createComponent } from "reakit-system";
+import { SELECT_KEYS } from "./__keys";
+import { useComposite, usePopoverDisclosure } from "reakit";
+
+export type SelectInputOptions = Pick<
+  SelectStateReturn,
+  | "toggle"
+  | "visible"
+  | "inputValue"
+  | "setInputValue"
+  | "removeSelected"
+  | "selected"
+>;
+
+const useSelectInput = createHook<SelectInputOptions, BoxHTMLProps>({
+  name: "SelectInput",
+  compose: [useComposite, usePopoverDisclosure],
+  keys: SELECT_KEYS,
+
+  useProps(
+    { selected, inputValue, visible, removeSelected, setInputValue },
+    { ref: htmlRef, ...htmlProps },
+  ) {
+    return {
+      type: "text",
+      value: inputValue,
+      onKeyDown: e => {
+        if (e.key === "Backspace" && inputValue === "") {
+          removeSelected(selected[selected.length - 1]);
+        }
+      },
+      onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
+        setInputValue(e.target.value);
+      },
```

Comment:
If you are not experimenting, make them perfect as Reakit's way.

get htmlOnChange and htmlKeyDown, prevent default and disabled handle everything.